### PR TITLE
added-calculator-files

### PR DESCRIPTION
**Pull request by Rehan-Khan-Engineer**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+function division() {
+  const initialResult = currentResult;
+  currentResult = currentResult / getUserInput();
+  createOutput(initialResult, "/", getUserInput());
+  writeToLog("DIVIDE", initialResult, getUserInput(), currentResult);
+}
+
+addBtn.addEventListener("click", addition);
```
Comment (anask601) [inline script.js:65]: addBtn.addEventListener("click", addition());  
what will happen if i write like this @Rehan-Khan-Engineer

**Review by syedabdullahali — APPROVED** [review]

File: script.js (hunk 1 above)
Comment (Rehan-Khan-Engineer) [reply script.js:65]: I think if you add parenthesis it means you are calling the function right then and there! 
A function name without parenthesis represents you are only registering  it and when the click occurs then only it gets executed. @anask601

Comment (anask601) [reply script.js:65]: Great @Rehan-Khan-Engineer

File: script.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+let defaultResult = 0;
+
+let currentResult = defaultResult;
+
+let logEntries = [];
```
Comment (anask601) [inline script.js:5]: can you use const here ?

File: script.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+function getUserInput() {
+  return parseInt(userInput.value);
```
Comment (anask601) [inline script.js:8]:
```jsx
parseInt(userInput.value);
```
what it is meaning of this line ?

File: index.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link rel="stylesheet" href="style.css" />
+    <script src="vendor.js" defer></script>
+    <script src="script.js" defer></script>
```
Comment (Rehmatulla-Khan) [inline index.html:9]: can you explain the defer here @Rehan-Khan-Engineer

File: script.js (hunk 2 above)
Comment (Rehan-Khan-Engineer) [reply script.js:5]: is this related to primitive type vs reference type ? I haven't studied it yet. will see in future. @anask601

File: script.js (hunk 3 above)
Comment (Rehan-Khan-Engineer) [reply script.js:8]: User input is always a string to JavaScript, thus to convert them into numbers we use parseInt, which is a built in js function. @anask601

File: index.html (hunk 4 above)
Comment (Rehan-Khan-Engineer) [reply index.html:9]: defer tells the browser to download the scripts right away when they are met, but DO NOT BLOCK PARSING HTML & execute scripts only after everything in HTML has been parsed! @Rehmatulla-Khan

**Review by anask601 — APPROVED** [review]

**Comment by Rehmatulla-Khan** [comment]
@Rehan-Khan-Engineer approving

**Comment by Rehan-Khan-Engineer** [comment]
> @Rehan-Khan-Engineer approving

@Rehmatulla-Khan  when?

**Pull request merged**